RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractWebAccountType.java

```diff
@@ -0,0 +1,369 @@
+import java.util.logging.Level;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProgress;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+
+/**
+ * Attempts to determine a user's role on a domain based on the URL.
+ */
+class ExtractWebAccountType extends Extract {
+
+    private static final Logger logger = Logger.getLogger(ExtractWebAccountType.class.getName());
+    
+    ExtractWebAccountType() {
+        moduleName = NbBundle.getMessage(ExtractWebAccountType.class, "ExtractWebAccountType.moduleName.text");
+    }
+    
+    private void extractDomainRoles(Content dataSource, IngestJobContext context) {
+        try {
+            // Get web history blackboard artifacts
+            Collection<BlackboardArtifact> listArtifacts = currentCase.getSleuthkitCase().getBlackboard().getArtifacts(
+                    Arrays.asList(new BlackboardArtifact.Type(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_HISTORY)),
+                    Arrays.asList(dataSource.getId()));
+            logger.log(Level.INFO, "Processing {0} blackboard artifacts.", listArtifacts.size()); //NON-NLS
+
+            // Set up collector for roles
+            RoleProcessor roleProcessor = new RoleProcessor(context);
+            
+            // Process each URL
+            for (BlackboardArtifact artifact : listArtifacts) {
+                if (context.dataSourceIngestIsCancelled()) {
+                    return;
+                }
+                
+                findRolesForUrl(artifact, roleProcessor);
+            }
+            
+            // Create artifacts
+            roleProcessor.createArtifacts();
+            
+        } catch (TskCoreException e) {
+            logger.log(Level.SEVERE, "Encountered error retrieving artifacts for domain role analysis", e); //NON-NLS
+        }
+    }
+    
+    /**
+     * Extract and store any role found in the given artifact.
+     * 
+     * @param artifact      The original artifact
+     * @param roleProcessor Object to collect and process domain roles.
+     * 
+     * @throws TskCoreException 
+     */
+    private void findRolesForUrl(BlackboardArtifact artifact, RoleProcessor roleProcessor) throws TskCoreException {
+        
+        BlackboardAttribute urlAttr = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL));
+        if (urlAttr == null) {
+            return;
+        }
+
+        BlackboardAttribute domainAttr = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN));
+        if (domainAttr == null) {
+            return;
+        }
+
+        String url = urlAttr.getValueString();
```

Comment:
Could this be lower-cased since the contains checks are looking for lower-cased strings? Unless there is a reason this must be case sensitive.